### PR TITLE
Add FluentBitIdleInputPlugins alert

### PR DESCRIPTION
**Pull request by vlvasilev**
**How to categorize this PR?**

/area logging
/kind enhancement

**What this PR does / why we need it**:
Recently we found that sometimes flued-bit pods stuck. They stops to prosses logs.
It is difficult to catch when this happens because the pods are still using CPU and Memory and looks like other working fluent-bit pods. The only difference is in the metric `fluentbit_input_bytes_total` which doesn't increase.
The new alert `FluentBitIdleInputPlugins` warn us when this is happening.
 
**Which issue(s) this PR fixes**:
This new alert will help debugging  [#94](https://github.com/gardener/logging/issues/94)

**Special notes for your reviewer**:

**Release note**:

```other operator
New alert `FluentBitIdleInputPlugins` for idle fluent-bit pods.
```

### CODE REVIEW

File: charts/seed-bootstrap/aggregate-prometheus-rules/fluent-bit.rules.yaml
Comment (vpnachev) [inline charts/seed-bootstrap/aggregate-prometheus-rules/fluent-bit.rules.yaml:30]: Hm, isn't 6 hours too long?

**Review by vpnachev — COMMENTED** [review]
/invite @wyb1 @istvanballok

File: charts/seed-bootstrap/aggregate-prometheus-rules/fluent-bit.rules.yaml
Comment (vlvasilev) [reply charts/seed-bootstrap/aggregate-prometheus-rules/fluent-bit.rules.yaml:30]: I don't know. There is two cases when this alert can be triggered without hitting this bug.
1. Non of the monitored component is logging a single line for a period of time.
    There was a similar alert which was triggered on dev and staging when there was a seed with one or few clusters(which were in hibernated state) and there was nothing to be logged on one of its node.
    The systemd input plugin should produce some logs after all but the question is for how long we have to wait.
 2. The node is under  high pressure and the input buffer is full. When the output plugging is processing the logs too slow the inputs are blocked until there is some free space in the buffer. In this case we have either a structural defect in the fluent-bit flow or we face network outage or massive Loki down event.
 
 So I don't have any evidence what will be the proper time.
 I am open for proposals.

Comment (vpnachev) [reply charts/seed-bootstrap/aggregate-prometheus-rules/fluent-bit.rules.yaml:30]: My point is that for 6 hours anything can happen and something else can restart stuck pod and you will not find it. Of course, the disadvantage of shorter firing times is that more false positive alerts might be generated.

Personally, I have no hard opinion.

**Review by vpnachev — APPROVED** [review]
/lgtm
/needs second-opinion

**Review by rfranzke — APPROVED** [review]
Personally, I'm fine with the alert for now (it seems there is no clear argument for or against the `6h`, so I guess we can start with it and change/adapt later if we see need).
/lgtm

However, I'm somewhat worried about

> Recently we found that sometimes flued-bit pods stuck. They stops to prosses logs.
It is difficult to catch when this happens because the pods are still using CPU and Memory and looks like other working fluent-bit pods. The only difference is in the metric fluentbit_input_bytes_total which doesn't increase.

Is there an upstream issue about it or are the fluent-bit maintainers aware of it?

**Comment by vlvasilev** [comment]
> Is there an upstream issue about it or are the fluent-bit maintainers aware of it?

I have no evidence whether the cause for he issue is in the fluent-bit itself or in our plugin.
I am making a new plugin release which is using [pprof](https://golang.org/pkg/net/http/pprof/) HTTP server to give us information about goroutines, mutexes and so on.

**Review by wyb1 — APPROVED** [review]

**Pull request merged**
